docs/misc/prune-example.txt

```diff
@@ -76,18 +80,23 @@ Notes
 -----
 
 2015-12-31 is kept due to the --keep-daily 14 rule (because it is applied
-first), not due to the --keep-monthly rule.
+first), not due to the --keep-monthly or --keep-yearly rule.
 
-Because of that, the --keep-monthly 6 rule keeps Nov, Oct, Sep, Aug, Jul and
-Jun. December is not considered for this rule, because that backup was already
-kept because of the daily rule.
+The --keep-yearly 1 rule does not consider the December 31st backup because it
+has already been kept due to the daily rule. There are no backups available
+from previous years, so the --keep-yearly target of 1 backup is not satisfied.
+Because of this, the 2015-01-01 archive (the oldest archive available) is kept.
```

Comment:
What would happen for --keep-yearly 2? Would it keep 2015-01-01 and 2015-01-02?

Reply:
No, it would only keep 2015-01-01, because that is the oldest archive available. (In the example, the user started backups on 2015-01-01).

The only change from the previous behavior is this: If any rule can't match its retention target (in your example, 2 yearly archives), it will retain the oldest archive if no previous rule has retained it. In this same example, without this patch, borg would prune that archive, even with the `--keep-yearly 1` rule.

This will generally mean that the first archive the user ever made with borg is retained until it is expired because it has been replaced by an archive that matches the rules given.